src/tests/lvm/lvm_shared_mount/lvm_shared_mount_test.go

```diff
@@ -0,0 +1,265 @@
+	fileContent := "This is some test data."
+	md5FilePath1 := "/volume/md5sum1.txt"
+	md5FilePath2 := "/volume/md5sum2.txt"
+	combinedCmd1 := fmt.Sprintf(
+		"echo '%s' > %s && md5sum %s > %s",
+		fileContent,
+		filePath,
+		filePath,
+		md5FilePath1,
+	)
+
+	out, err := executeCommandInPod(podName1, combinedCmd1)
+	Expect(err).To(BeNil(), "error: %v, output: %s", err, out)
+
+	// Deploy second BusyBox pod to verify data
+	podName2 := "busybox-second"
+	deployBusyBoxPod(podName2, app.GetPvcName(), app.VolType)
+
+	combinedCmd2 := fmt.Sprintf(
+		"md5sum %s > %s",
+		filePath,
+		md5FilePath2,
+	)
+
+	out, err = executeCommandInPod(podName2, combinedCmd2)
+	Expect(err).To(BeNil(), "error: %v, output: %s", err, out)
+
+	// Compare MD5 checksums from both pods
+	md5sum1, err := executeCommandInPod(podName1, fmt.Sprintf("cat %s", md5FilePath1))
+	Expect(err).To(BeNil(), "error %v", err)
+
+	md5sum2, err := executeCommandInPod(podName2, fmt.Sprintf("cat %s", md5FilePath2))
+	Expect(err).To(BeNil(), "error %v", err)
+
+	Expect(md5sum1 == md5sum2).Should(BeTrue(), "MD5 verification failed. Data has been altered.")
+
+	// Check the status of both pods
+	for _, podName := range []string{podName1, podName2} {
+		logf.Log.Info(fmt.Sprintf("Checking %s pod status", podName))
+		phase, err := k8stest.GetPodStatusByPrefix(podName, common.NSDefault)
+		Expect(err).To(BeNil(), "GetPodStatusByPrefix got error %v", err)
+		Expect(phase == coreV1.PodRunning).Should(BeTrue(), fmt.Sprintf("%s pod is not in running state", podName))
+	}
+
+	// Clean up resources
+	cleanUpResources([]string{podName1, podName2}, app.GetPvcName())
+}
+
+func deployBusyBoxPod(podName, pvcName string, volType common.VolumeType) *coreV1.Pod {
+	args := []string{"sleep", "10000000"}
+	podContainer := coreV1.Container{
+		Name:            podName,
+		Image:           imageutils.GetE2EImage(imageutils.BusyBox),
+		ImagePullPolicy: coreV1.PullAlways,
+		Args:            args,
+	}
+
+	volume := coreV1.Volume{
+		Name: "ms-volume",
+		VolumeSource: coreV1.VolumeSource{
+			PersistentVolumeClaim: &coreV1.PersistentVolumeClaimVolumeSource{
+				ClaimName: pvcName,
+			},
+		},
+	}
+
+	podObj, err := k8stest.NewPodBuilder(podName).
+		WithName(podName).
+		WithNamespace(common.NSDefault).
+		WithRestartPolicy(coreV1.RestartPolicyNever).
+		WithContainer(podContainer).
+		WithVolume(volume).
+		WithVolumeDeviceOrMount(volType).Build()
+	Expect(err).ToNot(HaveOccurred(), "Generating pod definition, err: %v", err)
+	Expect(podObj).ToNot(BeNil(), "failed to generate pod definition")
+
+	_, err = k8stest.CreatePod(podObj, common.NSDefault)
+	Expect(err).ToNot(HaveOccurred(), "Creating pod, err: %v", err)
+
+	Eventually(func() bool {
+		return k8stest.IsPodRunning(podName, common.NSDefault)
+	}, k8stest.DefTimeoutSecs, "2s").Should(Equal(true))
+
+	logf.Log.Info(fmt.Sprintf("%s pod is running.", podName))
+	return podObj
+}
+
+func executeCommandInPod(podName, cmd string) (string, error) {
```

Comment:
This is a utility function and should live in the shared code

Reply:
Removed it, Used already present function from k8stest package, no need to create another one here.